common/web/gesture-recognizer/src/test/auto/headless/segmentation.js

```diff
@@ -201,5 +201,213 @@ describe("Segmentation", function() {
+
+      // This is the one that reports all of our async assertion failures.
+      return finalPromise.then(() => {
+        const holdSegment = spy.secondCall.args[0];
+
+        assert.equal(holdSegment.type, 'hold');
+        assert.isAtLeast(holdSegment.duration, endSample.t - startSample.t);
+        assert.equal(holdSegment.speed, 0);
+        assert.isUndefined(holdSegment.angle);
+        assert.isUndefined(holdSegment.direction);
+        assert.equal(holdSegment.peakSpeed, 0);
+
+        // Samples may be cloned when passed through the segmenter!
+        assert.deepEqual(holdSegment.initialCoord, startSample);
+        assert.deepEqual(holdSegment.lastCoord, endSample);
+      });
+    });
+  });
+
+  describe("Idealized 'move'", function() {
+    beforeEach(function() {
+      this.fakeClock = sinon.useFakeTimers();
+    })
+
+    afterEach(function() {
+      this.fakeClock.restore();
+    })
+
+    /**
+     * Builds a diagonal sequence traveling 40 pixels 's', 40 pixels 'e' in a perfect diagonal lock-step.
+     */
+    const buildSampleSequence = () => {
+      const startSample = {
+        targetX: 1,
+        targetY: 1,
+        t: 100
+      };
+
+      let samples = [startSample];
+
+      for(let i=1; i <=20; i++) {
+        samples.push({
+          targetX: startSample.targetX + 2 * i,
+          targetY: startSample.targetY + 2 * i,
+          t: startSample.t + 10 * i
+        });
+      }
+
+      return samples;
+    }
+
+    it("'move' segment recognition + resolution", async function() {
+      let spy = sinon.fake();
+
+      const segmenter = new PathSegmenter(PathSegmenter.DEFAULT_CONFIG, spy);
+
+      const samples = buildSampleSequence();
+
+      const samplePromises = samples.map((sample) => {
+        return timedPromise(() => {
+          segmenter.add(sample);
+        }, sample.t - samples[0].t);
+      });
+
+      // Find and replace select promises with then'd versions of themselves to 'hook in' as needed.
+      // We know the 'move' segment will appear upon completion of the first sample's Promise.
+
+      const segment2Recognition = sinon.fake();
+      const segment2Resolution  = sinon.fake();
+
+      samplePromises[0] = samplePromises[0].then(async () => {
+        assert.isAtLeast(spy.callCount, 2);
+        const moveSegment = spy.secondCall.args[0];
+
+        assert.equal(await promiseStatus(moveSegment.whenRecognized), PromiseStatuses.PROMISE_PENDING);
+        assert.equal(await promiseStatus(moveSegment.whenResolved), PromiseStatuses.PROMISE_PENDING);
+
+        // Now that we're sure they haven't already been called, let's set the `fake` calls in place.
+        moveSegment.whenRecognized.then(segment2Recognition);
+        moveSegment.whenResolved.then(segment2Resolution);
+
+        assert.isNotNull(moveSegment);
+      });
+
+      // Given the current engine setup, we know that after 10 samples, we pass the recognition threshold and
+      // thus should be classified as a 'move'.
```

Comment:
This highlights something that may need further examination - the move is quick enough that we should be able to trigger it earlier.  There's no checks for move-based triggering before the time threshold if there's no segmentation trigger yet, though.  Yet.

Reply:
And, yep.  Turned out to be a two-liner that I'd overlooked; found it pretty quickly this morning.  The related fix has already been implemented in the base branch.